packages/trade-widget/src/components/BuyWidget/BuyWidget.tsx

```diff
@@ -0,0 +1,207 @@
+    tokenBalances: AccountTokenBalances;
+    selectedToken: AssetToken;
+    web3Wrapper: Web3Wrapper;
+    zeroEx: ZeroEx;
+    dispatcher: Dispatcher;
+    quote: Quote;
+}
+
+interface BuyWidgetState {
+    amount?: BigNumber;
+}
+
+const ETH_DECIMAL_PLACES = 18;
+class BuyWidget extends React.Component<BuyWidgetProps, BuyWidgetState> {
+    private _forwarder: ForwarderWrapper;
+    constructor(props: any) {
+        super(props);
+        this.state = {
+            amount: new BigNumber(1),
+        };
+
+        this._handleAmountChange = this._handleAmountChange.bind(this);
+        this._handleSubmitAsync = this._handleSubmitAsync.bind(this);
+    }
+
+    public async componentDidUpdate() {
+        await this._quoteSelectedTokenAsync();
+    }
+
+    render() {
+        const { address, weiBalances, tokenBalances, selectedToken, quote } = this.props;
+        const tokenBalance = tokenBalances[address] ? tokenBalances[address][selectedToken] : new BigNumber(0);
+        const weiBalance = weiBalances[address] || new BigNumber(0);
+        return (
+            <Content>
+                <AccountBlockie
+                    account={address}
+                    weiBalance={weiBalance}
+                    tokenBalance={tokenBalance}
+                    selectedToken={selectedToken}
+                />
+                <Label isSize="small">SELECT TOKEN</Label>
+                <Field isFullWidth={true}>
+                    <TokenSelector
+                        selectedToken={this.props.selectedToken}
+                        onChange={this._handleTokenSelected.bind(this)}
+                    />
+                </Field>
+                <Label style={{ marginTop: 30 }} isSize="small">
+                    BUY AMOUNT
+                </Label>
+                <Field hasAddons={true}>
+                    <Control isExpanded={true}>
+                        <Input type="text" placeholder="1" onChange={this._handleAmountChange.bind(this)} />
+                    </Control>
+                    <Control>
+                        <Select>
+                            <option>ETH</option>
+                            <option>ZRX</option>
+                        </Select>
+                    </Control>
+                </Field>
+                {/* <Field>
+                    <strong> ESTIMATED COST </strong>
+                </Field> */}
+                <Field style={{ marginTop: 20 }}>
+                    <Button
+                        isLoading={!this._hasQuoteForSelectedToken()}
+                        isFullWidth={true}
+                        isColor="info"
+                        onClick={this._handleSubmitAsync}
+                    >
+                        BUY TOKENS
+                    </Button>
+                </Field>
+                <Field style={{ marginTop: 20 }} isGrouped={'centered'}>
+                    <img style={{ marginLeft: '0px', height: '20px' }} src="/images/powered.png" />
```

Comment:
For in-line, if it's `px` you want, you can simply omit the suffix: `height: 20, `